xprof/xprof.rb.in

```diff
@@ -237,25 +272,28 @@ class Sync_daemon
   end
 
   def finalize
-    LOGGER.info { "finalize Sync_daemon" }
-    lazy_exec do
+    lazy_exec('Finalize Sync_daemon') do
       `kill -#{RT_SIGNAL_FINISH} #{@pid}`
     end
   end
 
   def local_barrier(name)
-    LOGGER.info { "Local_barrier #{name}" }
-    lazy_exec do
+    lazy_exec("Local_barrier #{name}") do
       `kill -#{RT_SIGNAL_LOCAL_BARRIER} #{@pid}`
     end
   end
 
   def global_barrier
-    LOGGER.info { 'global_barrier' }
-    lazy_exec do
+    lazy_exec('Global_barrier') do
       `kill -#{RT_SIGNAL_GLOBAL_BARRIER} #{@pid}`
     end
   end
+
+  def self.open
+    yield f = new
+  ensure
+    f.finalize
+  end
```

Comment:
As someone new to ruby, this one is pretty subtle - comment would be helpful.

Reply:
Good point, will add!

Reply:
Pushed with 8b8162b